src/java/htsjdk/samtools/util/BlockCompressedInputStream.java

```diff
@@ -192,45 +203,45 @@ public int read(final byte[] buffer)
      * @
      */
     public String readLine() throws IOException {
-    	int available = available();
+        int available = available();
         if (available == 0) {
             return null;
         }
         if(null == buf){ // lazy initialisation 
-        	buf = new ByteArrayOutputStream(8192);
+            buf = new ByteArrayOutputStream(8192);
         }
         buf.reset();
-    	boolean done = false;
-    	boolean foundCr = false; // \r found flag
+        boolean done = false;
+        boolean foundCr = false; // \r found flag
         while (!done) {
-        	int linetmpPos = mCurrentOffset;
-        	int bCnt = 0;
-        	while((available-- > 0)){
-        		final byte c = mCurrentBlock[linetmpPos++];
-        		if(c == eol){ // found \n
-        			done = true;
-        			break;
-        		} else if(foundCr){  // previous char was \r
-        			--linetmpPos; // current char is not \n so put it back
-        			done = true;
-        			break;
-        		} else if(c == eolCr){ // found \r
-					foundCr = true;
-        			continue; // no ++bCnt
-        		}
-				++bCnt;
-        	}
-        	if(mCurrentOffset < linetmpPos){
-				buf.write(mCurrentBlock, mCurrentOffset, bCnt);
-	        	mCurrentOffset = linetmpPos;
-        	}
-        	available = available();    
-        	if(available == 0){
-        		// EOF
-        		done = true;
-        	}
-        }
-    	return buf.toString();
+            int linetmpPos = mCurrentOffset;
+            int bCnt = 0;
+            while((available-- > 0)){
```

Comment:
this line is so horrible `(available-- > 0)` - can you use this as a opportunity to make it cleaner?